### PR TITLE
Workaround for DashArray with starting 0 on canvas layers

### PR DESCRIPTION
**Pull request by McBen**
On svg layers a `dashArray="0 8 16 8"` is fine.  
`(example result: "  ----    ----  ")`
On canvas layers the internal `setLineDash([0,8,16,8])` will create artefacts at the begining of the dash. 
`(example result: "-  ----  -  ----  ")`

So we got different results on svg and canvas layers.

This can be workaround by converting the first zero argument into the dashOffset.

( required PR #7867 to support dashOffset)

### CODE REVIEW

**Comment by Falke-Design** [comment]
> ( required PR #7867 to support dashOffset)

This PR will already contains the changes of #7867

File: src/layer/vector/Canvas.js [hunk 1]
```diff
@@ -339,7 +339,13 @@ export var Canvas = Renderer.extend({
 
 		if (options.stroke && options.weight !== 0) {
 			if (ctx.setLineDash) {
-				ctx.setLineDash(layer.options && layer.options._dashArray || []);
+				ctx.lineDashOffset = Number(options.dashOffset || 0);
```
Comment (Falke-Design) [inline src/layer/vector/Canvas.js:342]: Is `Number` really neaded? We don't normaly use this in Leaflet

```suggestion
				ctx.lineDashOffset = options.dashOffset || 0;
```

**Comment by Falke-Design** [comment]
@McBen can you please provide a sample. I can't reproduce the artefact-differences from canvas and svg.
Here a sample: https://plnkr.co/edit/dwb7kGpUrwA0xwsH with 1.7.1

This `artefacts` are because of linecap: 
 - https://plnkr.co/edit/Wltt2AmC7eG0qN9p
 - https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineCap

**Comment by McBen** [comment]
yes, you're right. It's about the linecap.

**Pull request closed without merging**